core/domain/user_services.py

```diff
@@ -920,6 +941,28 @@ def has_fully_registered_account(user_id):
         feconf.REGISTRATION_PAGE_LAST_UPDATED_UTC)
 
 
+def get_all_profiles_auth_details_by_parent_user_id(parent_user_id):
+    """Gets domain objects representing the auth details for all profiles
+    associated with the user having the given parent_user_id.
+
+    Args:
+        parent_user_id: str. User id of the parent_user whose associated
+            profiles we are querying for.
+
+    Returns:
+        list(UserAuthDetails). The UserAuthDetails domain objects
+        corresponding to the profiles linked to given parent_user_id. If that
+        parent user does not have any profiles linked to it, the
+        returned list will be empty.
+    """
+    get_auth_details_by_user_id(parent_user_id, strict=True)
```

Comment:
Why is this here? It's not used?

(If it is supposed to be here, add a comment explaining what it's for. It may make more sense to have a new function like require_existence_of_user_id() since GET semantics are a bit weird here since you don't actually use the value.)

Reply:
I think this was the only place where we wanted to keep this check (to ensure parent user exists before making such a query) so I did not make a new function and instead used the existing storage class function only.
Note: Even if we pass a non-existing parent_user id to the `UserAuthDetailsModel.get_all_profiles_by_parent_user_id` method, we will get an empty list only, but acc to me raising an error seems imp.